fifteen_min/src/find_amenities.rs

```diff
@@ -0,0 +1,143 @@
+use map_gui::tools::{ChooseSomething, ColorLegend};
+use map_gui::ID;
+use map_model::AmenityType;
+use widgetry::{
+    Cached, Choice, Color, Drawable, EventCtx, GfxCtx, HorizontalAlignment, Key, Line, Panel,
+    SimpleState, State, Transition, VerticalAlignment, Widget,
+};
+
+use crate::isochrone::{Isochrone, Options};
+use crate::viewer::{draw_star, draw_unwalkable_roads, HoverKey, HoverOnBuilding};
+use crate::App;
+
+/// Calculate isochrones around each amenity on a map and merge them together using the min value
+pub struct FindAmenity {
+    options: Options,
+}
+
+impl FindAmenity {
+    pub fn new(ctx: &mut EventCtx, options: Options) -> Box<dyn State<App>> {
+        ChooseSomething::new(
+            ctx,
+            "Choose an amenity",
+            AmenityType::all()
+                .into_iter()
+                .map(|at| Choice::new(at.to_string(), at))
+                .collect(),
+            Box::new(move |at, ctx, app| {
+                let multi_isochrone = create_multi_isochrone(ctx, app, at, options.clone());
+                return Transition::Push(Results::new(ctx, app, multi_isochrone, at));
+            }),
+        )
+    }
+}
+
+/// For every one of the requested amenity on the map, draw an isochrone
+fn create_multi_isochrone(
+    ctx: &mut EventCtx,
+    app: &App,
+    category: AmenityType,
+    options: Options,
+) -> Isochrone {
+    let map = &app.map;
+    // For a category, find all matching stores
+    let mut stores = Vec::new();
+    for b in map.all_buildings() {
+        if b.has_amenity(category) {
+            stores.push(b.id);
+        }
+    }
+    Isochrone::new(ctx, app, stores, options.clone())
+}
+
+struct Results {
+    draw: Drawable,
+    isochrone: Isochrone,
+    hovering_on_bldg: Cached<HoverKey, HoverOnBuilding>,
+    // TODO Can't use Cached due to a double borrow
+    hovering_on_category: Option<(AmenityType, Drawable)>,
```

Comment:
This and `draw_unwalkable_roads` were copied from the main viewer. I think we can remove both from the results screen for amenity search. The first one would let you hover on a category in the panel and draw matching stores on the map, but now we draw stars around them and there's only one category shown.